backend/src/load.rs

```diff
@@ -0,0 +1,404 @@
+use std::collections::{HashMap, HashSet};
+use std::cmp::{Ordering, max};
+use std::io::Read;
+
+use chrono::{Duration, NaiveDateTime};
+use json::{self, JsonValue};
+
+use util::start_idx;
+use errors::*;
+
+const WEEKS_IN_SUMMARY: usize = 12;
+
+/// Contains the data loaded from the file system.
+/// Useful to avoid passing around each individual field, instead passing the
+/// entire struct.
+pub struct ServerData {
+    pub summary_rustc: Summary,
+    pub summary_benchmarks: Summary,
+
+    /// A set containing all crate names.
+    pub crate_list: HashSet<String>,
+
+    /// A set containing all phase names, across all crates.
+    pub phase_list: HashSet<String>,
+
+    /// TODO: Better docs. It's unknown how this is used on the client.  A set
+    /// of test names. Test names are found from the file path, everything
+    /// before the `--` is included.
+    pub benchmarks: HashSet<String>,
```

Comment:
I'm certain that for all three we get duplicate entries: phases are added into whenever we encounter a phase across all files, crates are added into on encountering a crate [and we test the same crates], benchmarks are also added repeatedly.

We don't use these values on the backend side near as I can tell other than returning them in the `/info` getter.